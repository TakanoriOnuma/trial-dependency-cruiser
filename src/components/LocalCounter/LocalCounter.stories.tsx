import { StoryObj, Meta } from '@storybook/react';
import { within, userEvent } from '@storybook/test';

import { LocalCounter } from './LocalCounter';

type Story = StoryObj<typeof LocalCounter>;

const meta: Meta<typeof LocalCounter> = {
  title: 'LocalCounter',
  component: LocalCounter,
};

export default meta;

export const Base: Story = {};

export const Play: Story = {
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);
    await userEvent.click(canvas.getByRole('button', { name: '+' }));
    await userEvent.click(canvas.getByRole('button', { name: '+' }));
    await userEvent.click(canvas.getByRole('button', { name: '+' }));
  },
};

export const InitialCount: Story = {
  args: {
    initialCount: 10,
  },
};
