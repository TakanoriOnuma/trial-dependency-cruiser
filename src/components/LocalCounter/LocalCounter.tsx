import { FC } from 'react';

import { Counter, useCounter } from '../Counter';
// こっちの書き方だと循環参照になる
// import { Counter, useCounter } from '~/components';

export type LocalCounterProps = {
  initialCount?: number;
};

export const LocalCounter: FC<LocalCounterProps> = ({ initialCount }) => {
  const { count, increment, decrement } = useCounter(initialCount);

  return (
    <Counter count={count} onIncrement={increment} onDecrement={decrement} />
  );
};
