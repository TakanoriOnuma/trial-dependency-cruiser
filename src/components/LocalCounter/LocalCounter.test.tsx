import { composeStories } from '@storybook/react';
import * as stories from './LocalCounter.stories';
import { render, screen } from '@testing-library/react';

const { Base, InitialCount, Play } = composeStories(stories);

describe('LocalCounter', () => {
  test('Base', () => {
    const { container } = render(<Base />);
    expect(container).toMatchSnapshot();
  });

  test('InitialCount', () => {
    const { container } = render(<InitialCount />);
    expect(container).toMatchSnapshot();
    expect(screen.getByText('10')).toBeInTheDocument();
  });

  test('Play', async () => {
    const { container } = render(<Play />);

    await Play.play?.({ canvasElement: container });
    expect(container).toMatchSnapshot();
    expect(screen.getByText('3')).toBeInTheDocument();
  });
});
