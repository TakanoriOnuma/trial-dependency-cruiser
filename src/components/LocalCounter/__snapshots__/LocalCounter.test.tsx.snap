// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LocalCounter > Base 1`] = `
<div>
  <div
    style="display: flex;"
  >
    <button>
      -
    </button>
    <span
      style="padding: 0px 5px;"
    >
      0
    </span>
    <button>
      +
    </button>
  </div>
</div>
`;

exports[`LocalCounter > InitialCount 1`] = `
<div>
  <div
    style="display: flex;"
  >
    <button>
      -
    </button>
    <span
      style="padding: 0px 5px;"
    >
      10
    </span>
    <button>
      +
    </button>
  </div>
</div>
`;

exports[`LocalCounter > Play 1`] = `
<div>
  <div
    style="display: flex;"
  >
    <button>
      -
    </button>
    <span
      style="padding: 0px 5px;"
    >
      3
    </span>
    <button>
      +
    </button>
  </div>
</div>
`;
